src/game/client/draw/w3dmodeldraw.cpp

```diff
@@ -13,3 +13,1149 @@
+bool Does_State_Exist(std::vector<ModelConditionInfo> const &v, BitFlags<MODELCONDITION_COUNT> const &f)
+{
+    for (auto &i : v) {
+        for (int a = i.Get_Conditions_Count() - 1; a >= 0; a--) {
+            if (f == i.Get_Conditions_Yes(a)) {
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+uint64_t Make_Transition(int old_state, int new_state)
+{
+    return new_state | (uint64_t)((int64_t)old_state << 32);
+}
+
+void Parse_Bone_Name_Key(INI *ini, void *instance, void *store, void const *user_data)
+{
+    Utf8String str(ini->Get_Next_Token());
+    str.To_Lower();
+
+    if (instance) {
+        ((ModelConditionInfo *)instance)->Add_Public_Bone(str);
+    }
+
+    if (str.Is_Empty() || str.Is_None()) {
+        *(NameKeyType *)store = NAMEKEY_INVALID;
+    } else {
+        *(NameKeyType *)store = g_theNameKeyGenerator->Name_To_Key(str);
+    }
+}
+
+void Parse_Show_Hide_Sub_Object(INI *ini, void *instance, void *store, void const *user_data)
+{
+    std::vector<ModelConditionInfo::HideShowSubObjInfo> *v = (std::vector<ModelConditionInfo::HideShowSubObjInfo> *)store;
+    Utf8String str = ini->Get_Next_Ascii_String();
+    str.To_Lower();
+
+    if (str.Is_None()) {
+        v->clear();
+    } else {
+        while (str.Is_Not_Empty()) {
+            bool found = false;
+
+            for (auto &i : *v) {
+                if (!strcasecmp(i.name, str)) {
+                    i.visible = (uintptr_t)user_data != 0;
+                    found = true;
+                }
+            }
+
+            if (!found) {
+                ModelConditionInfo::HideShowSubObjInfo info;
+                info.name = str;
+                info.visible = (uintptr_t)user_data != 0;
+                v->push_back(info);
+            }
+
+            str = ini->Get_Next_Ascii_String();
+            str.To_Lower();
+        }
+    }
+}
+
+void Parse_Weapon_Bone_Name(INI *ini, void *instance, void *store, void const *user_data)
+{
+    Utf8String *str = (Utf8String *)store;
+    int index = ini->Scan_IndexList(ini->Get_Next_Token(), s_theWeaponSlotTypeNames);
+    str[index] = ini->Get_Next_Ascii_String();
+    str[index].To_Lower();
+
+    if (str[index].Is_None()) {
+        str[index].Clear();
+    }
+
+    if (instance) {
+        ((ModelConditionInfo *)instance)->Add_Public_Bone(str[index]);
+    }
+}
+
+void Parse_Animation(INI *ini, void *instance, void *store, void const *user_data)
+{
+    ModelConditionInfo *minfo = (ModelConditionInfo *)instance;
+    Utf8String name = ini->Get_Next_Ascii_String();
+    name.To_Lower();
+    const char *str = ini->Get_Next_Token_Or_Null();
+    float distance;
+
+    if (str) {
+        distance = ini->Scan_Real(str);
+    } else {
+        distance = 0.0f;
+    }
+
+    captainslog_dbgassert((uintptr_t)user_data != 1 || distance == 0.0f,
+        "You should not specify nonzero DistanceCovered values for Idle Anims");
+
+    str = ini->Get_Next_Token_Or_Null();
+    int count;

```

Comment:
This can be greatly simplified as

`int count = std::min(str ? ini->Scan_Int(str) : 1, 1);`

Reply:
fixed.